provider.go

```diff
@@ -287,16 +288,42 @@ func (g *InstanceGroup) Increase(
 	ctx context.Context,
 	delta int,
 ) (succeeded int, err error) {
-	servers := make([]*cloudscale.Server, 0, delta)
-	errs := make([]error, 0)
-
 	publicKey, err := g.publicKey()
 	if err != nil {
 		return 0, err
 	}
 
 	tagMap := g.tagMap()
 
+	// Channel for server uuids
+	servers := make(chan string)
+	// Because concurency, we use a channel for collecting errors, as
+	// slice.append is not thread safe
+	errors_c := make(chan error)
+
+	// Await all servers being ready
+	var wg sync.WaitGroup
+	go func(servers <-chan string) {
```

Comment:
This goroutine needs a `wg.Add(1)` / `wg.Done()` combo as well, as you otherwise can run into the race condition where the first server has not been popped out of the channel yet, when `wg.Wait()` is called below (I actually observed this on a test host).

Reply:
done